c10/cuda/CUDACachingAllocator.h

```diff
@@ -118,9 +118,17 @@ struct SegmentInfo {
   std::vector<BlockInfo> blocks;
 };
 
+struct AllocFreeEvent {
+  intptr_t ptr; // start location in memory
+  size_t size; // size in bytes
+  bool type; // 0 for free and 1 for allocate
+  int device;
+};
+
 C10_CUDA_API void* raw_alloc(size_t nbytes);
 C10_CUDA_API void* raw_alloc_with_stream(size_t nbytes, cudaStream_t stream);
 C10_CUDA_API void raw_delete(void* ptr);
+C10_CUDA_API std::vector<std::vector<AllocFreeEvent>> GetAllocFreeEvents();
```

Comment:
for camel naming, it should be `getAllocFreeEvents`

Reply:
Done. Thank you.